### ISSUE TITLE
VS 'Send to interactive' displays warnings (Invalid directive #interactiveprompt)

### ISSUE THREAD

**Issue by abonie** [dotnet/fsharp#14433]
"Send to interactive" (hitting Alt+Enter with some code selected to send it to VS-FSI) displays warnings for all inputs

Example output after trying to send something to VS-FSI
```
Welcome to F# Interactive for .NET Core in Visual Studio. To execute code, either
  1. Use 'Send to Interactive' (Alt-Enter or right-click) from an F# script. The F# Interactive process will
     use any global.json settings associated with that script.
  2. Press 'Enter' to start. The F# Interactive process will use default settings.
> 

Microsoft (R) F# Interactive version 12.4.0.0 for F# 7.0
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> val it: unit = ()

> 
  #interactiveprompt "hide"
  ^^^^^^^^^^^^^^^^^^^^^^^^^

stdin(3,1): warning FS3353: Invalid directive '#interactiveprompt hide'

val it: int list = [1; 2; 3; 4]

> 
  System.IO.File.WriteAllText(@"C:\Users\adboniec\AppData\Local\Temp\tmpA455.tmp.pid", string (System.Diagnostics.Process.GetCurrentProcess().Id));;
  ^^^^^^^^^^^^^^^^^^^^^^^^^

stdin(1,1): warning FS3353: Invalid directive '#interactiveprompt show'
```

**Related information**

Provide any related information (optional):

* Editing Tools: Visual Studio Version 17.5.0 Preview 3.0 [33202.356.main]

**Comment by abonie** [comment]
Introduced by this PR https://github.com/dotnet/fsharp/pull/14383 most likely

**Comment by abonie** [comment]
It seems to me that VS uses FSI that does not have changes from #14383 , but I don't know how to verify this or if that's even possible. @KevinRansom thoughts?

**Comment by KevinRansom** [comment]
The fsi has flowed.

**Comment by yuuuxt** [comment]
Using 17.5.0 (not a preview) and is having this issue. Shall we wait for a fix, or it's already been fixed?
![图片](https://user-images.githubusercontent.com/3153534/221348149-a45f73c9-2bf0-43ef-8109-4a99bf83915a.png)

**Comment by vzarytovskii** [comment]
@KevinRansom could you please look whether change made it to 17.6 only?

**Comment by halcwb** [comment]
@vzarytovskii and @KevinRansom 

Never had this problem, now with update to the current version, I get this problem.

Microsoft Visual Studio Community 2022
Version 17.5.2
VisualStudio.17.Release/17.5.2+33502.453
Microsoft .NET Framework
Version 4.8.09032

Installed Version: Community

ASP.NET and Web Tools   17.5.317.37931
ASP.NET and Web Tools

Azure App Service Tools v3.0.0   17.5.317.37931
Azure App Service Tools v3.0.0

C# Tools   4.5.0-6.23128.14+651d32d8017cfcc60fcbd67383601fa9a3deb145
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

ILSpy.AddIn   1.0
Integration of the ILSpy Decompiler into Visual Studio.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

NuGet Package Manager   6.5.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

TypeScript Tools   17.0.20105.2003
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.5.0-6.23128.14+651d32d8017cfcc60fcbd67383601fa9a3deb145
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.5.0-beta.23053.5+794b7c259d9646a7eb685dad865aa27da7940a21
Microsoft Visual F# Tools

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

**Comment by vzarytovskii** [comment]
It might've been fixed only in 17.6.

**Comment by KevinRansom** [comment]
I have verified this in 17.5.2, it works correctly:
![image](https://user-images.githubusercontent.com/5175830/226453092-3d475298-1c0c-4cb0-9dbc-2d881763d0ca.png)

**Comment by KevinRansom** [comment]
It is possible that VS is picking up a global.json file that pins the dotnet SDK to a version earlier than 17.5.2.  Although the 17.5.0 release SDK should also work correctly.

To verify whether something like this is happening:
in the fsi script type this:

````
let x = System.Diagnostics.Process.GetCurrentProcess()
for item in x.Modules do
    if item.FileName.Contains("fsi.dll") then
        printfn "%A" item.FileName
````
![image](https://user-images.githubusercontent.com/5175830/226455885-9ae245bf-b2f6-4184-b6e6-139737013a6d.png)

**Comment by isaacabraham** [comment]
I'm seeing this in the latest public VS (17.5.3) as well. Haven't used it in a file but it's there now.

**Comment by yuuuxt** [comment]
At least I found how to avoid it, an example here to help debug: 
- `reset interactive session`, directly run `1+1;;` in interactive, showing `F# Interactive version 12.5.0.0 for F# 7.0` and is working fine;
- `reset interactive session`, open an unsaved new `fsx` file , write `1+1` in it and `execute in interactive`, showing `F# Interactive version 12.5.0.0 for F# 7.0` and it's working fine;
- `reset interactive session`, open a saved `fsx` file in a project (whose `TargetFramework` is `net6.0`) , write `1+1` in it and `execute in interactive`, showing `F# Interactive version 12.0.1.0 for F# 6.0` and showing error as above.

**Comment by KevinRansom** [comment]
Okay I just validated the fix, using 17.7 preview 2.

![image](https://user-images.githubusercontent.com/5175830/236405646-dc69bd48-dd63-4dd3-b98e-58b20ea2908a.png)
